vectordraw/vectordraw.py

```diff
@@ -0,0 +1,387 @@
+            'vector_properties_label': self.vector_properties_label,
+            'background': self.background,
+            'vectors': self.get_vectors,
+            'points': self.get_points,
+            'expected_result': self.get_expected_result
+        }
+
+    @property
+    def user_state(self):
+        """
+        Return user state, which is a combination of most recent answer and result.
+        """
+        user_state = self.answer
+        if self.result:
+            user_state['result'] = self.result
+        return user_state
+
+    @property
+    def background(self):
+        """
+        Return information about background to draw for this exercise.
+        """
+        return {
+            'src': self.background_url,
+            'width': self.background_width,
+            'height': self.background_height,
+        }
+
+    @property
+    def get_vectors(self):
+        """
+        Load info about vectors for this exercise from JSON string specified by course author.
+        """
+        return json.loads(self.vectors)
+
+    @property
+    def get_points(self):
+        """
+        Load info about points for this exercise from JSON string specified by course author.
+        """
+        points = json.loads(self.points)
+        for point in points:
+            # If author did not specify whether point should be drawn in fixed location (True)
+            # or placed by student (False), we default to True;
+            # template needs this info to determine whether it should add option
+            # for selecting point to dropdown menu:
+            if 'fixed' not in point:
+                point['fixed'] = True
+        return points
+
+    @property
+    def get_expected_result(self):
+        """
+        Load info about expected result for this exercise
+        from JSON string specified by course author.
+        """
+        return json.loads(self.expected_result)
+
+    def student_view(self, context=None):
+        """
+        The primary view of the VectorDrawXBlock, shown to students
+        when viewing courses.
+        """
+        context['self'] = self
+        fragment = Fragment()
+        fragment.add_content(loader.render_template('static/html/vectordraw.html', context))
+        fragment.add_css_url(
+            "//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.3.0/css/font-awesome.min.css"
+        )
+        fragment.add_css(loader.load_unicode('static/css/vectordraw.css'))
+        # Workbench doesn't have Underscore.js, so add it:
+        if WorkbenchRuntime and isinstance(self.runtime, WorkbenchRuntime):
+            fragment.add_javascript_url(
+                "//cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.2/underscore-min.js"
+            )
+        fragment.add_javascript_url(
+            "//cdnjs.cloudflare.com/ajax/libs/jsxgraph/0.98/jsxgraphcore.js"
+        )
+        fragment.add_javascript(loader.load_unicode("static/js/src/vectordraw.js"))
```

Comment:
@smarnach Just FYI, I ended up getting rid of the `static` directory entirely -- it seemed weird to have JS and CSS files spread out over two directories. So both CSS and JS files for this XBlock now live in `public/{css,js}`, and templates are stored in `templates/html`. This is mimics the directory layout of, e.g., problem-builder and xblock-drag-and-drop-v2.